### ISSUE TITLE
Some question about the class 'DilatedResnet' in deeplab.py

### ISSUE THREAD

**Issue by BruceHsuan** [1991viet/Deeplab-pytorch#2]
In the class 'DilatedResnet' in deeplab.py ,there exist expressions : 
`self.layer1 = self._make_layer(block, 64, layers[0], stride=1, dilation=1)`
`self.layer2 = self._make_layer(block, 128, layers[1], stride=2, dilation=1)`
` self.layer3 = self._make_layer(block, 256, layers[2], stride=1, dilation=2)`
` self.layer4 = self._make_layer(block, 512, layers[3], stride=1, dilation=4)`
I think stride in layer3,4 should be 2 other than 1,because in the paper ,conv3_x has output size 28*28, conv4_x has output size 14 * 14, conv5_x has output size 7 * 7.But in your implementation ,  the strides are 1. I am puzzled about it.
Can any explanation be provided? thx

**Comment by 1991viet** [comment]
Hi,
Thanks for your question. I guess you forget the parameter dilation. For example, In convolution layer, stride 1, pad 1 (both sides), filter size is 3 and default dilation 1 keeps output to have the same size as input. But if dilation is 2, output size is reduced by a half !
Bests,

**Comment by BruceHsuan** [comment]
Hi,
Thanks for your response.
I don't think setting dilation be 2 can reduce the output size by half. I just did such a experiment
```
dilation = 2
conv = nn.Conv2d(64, 128, kernel_size=3, stride=1,
                       padding=dilation, dilation=dilation, bias=False)
x = torch.Tensor(50, 64, 256, 256)
y = conv(x)
```
I run the code above, input the x into conv with stride 1 dilation 2,and the output y seems to be a [50, 128, 256, 256] tensor, proving that setting dilation to be 2 cannot reduce the output size by half

I give my calculation as support:
To get a dilated convolution filter, we can insert (dilation-1) 0s between each pair of neighboring elements in a convolution filter,that means the effective size of a size (3,3) dilated convolution filter is 3+(3-1)(dilation-1).If the size of input is [H,H],  padding is dilation, so the output height H_new = [H -(3 + 2*(dilation-1) )+ 2 * dilation] /stride+1 = [H-1]/stride+1
( sorry, I don't know how to use latex here
Bests,

**Comment by 1991viet** [comment]
Hi,
Sorry for my misremembering. You are right, the dilation does not reduce output size by half as I said. I coded this project a couple of months ago, hence there are some parts I forget. Since now I am working, do you mind if I answer you on the weekend (I will try to answer asap) ? But I could make you sure that my code followed strictly the paper, and output showed on README file was what we had after training the model.
Bests,

**Comment by BruceHsuan** [comment]
Thank you！your codes help a lot

**Comment by 1991viet** [comment]
Thank you :) I will answer your question by Sunday.

**Comment by BruceHsuan** [comment]
Sorry for responding too late.Thank U for your explanation. I've got it . I used to feel confused that your implementation of Resnet differs from the official version. Now I understand you designed the 'modified' Resnet. I am going to compare the result of these two versions.Thank U a lot!! :)
